domain/ide/command/folder_handler.go

```diff
@@ -139,3 +143,85 @@ func GetTrustMessage(untrusted []types.Folder) string {
 		"the package manager to get dependency information. You should only scan folders you trust."+
 		"\n\nUntrusted Folders: \n%s\n\n", untrustedFolderString)
 }
+
+func UpdateFolderConfigOrg(c *config.Config, storedConfig *types.FolderConfig, folderConfig *types.FolderConfig) {
+	// For configs that have been migrated, we use the org returned by LDX-Sync unless the user has set one.
+	if storedConfig.OrgMigratedFromGlobalConfig {
+		if folderConfig == nil {
```

Comment:
You can use this: `LSPInitialized` (set at application/server/server.go:393) to check successful and finished initialization.